src/components/dashboard/data/utils.js

```diff
@@ -1,4 +1,33 @@
-import { ASSIGNMENT_TYPES } from '../../enterprise-user-subsidy/enterprise-offers/data/constants';
+import { ASSIGNMENT_TYPES, ASSIGNMENT_ACTION_TYPES } from '../../enterprise-user-subsidy/enterprise-offers/data/constants';
+import {
+  LEARNER_ACKNOWLEDGED_ASSIGNMENT_CANCELLATION_ALERT,
+  LEARNER_ACKNOWLEDGED_ASSIGNMENT_EXPIRATION_ALERT,
+} from '../main-content/course-enrollments/data/constants';
+
+export function getIsActiveExpiredAssignment(assignments) {
+  const lastExpiredAlertDismissedTime = global.localStorage.getItem(LEARNER_ACKNOWLEDGED_ASSIGNMENT_EXPIRATION_ALERT);
+
+  const activeExpiredAssignments = assignments.filter((assignment) => (
+    assignment?.actions.some((action) => (
+      action.actionType === ASSIGNMENT_ACTION_TYPES.AUTOMATIC_CANCELLATION_NOTIFICATION
+      && new Date(action.completedAt) > new Date(lastExpiredAlertDismissedTime)
+    ))
+  ));
+  return activeExpiredAssignments.length > 0;
+}
+
+export function getIsActiveCancelledAssignment(assignments) {
```

Comment:
nit: `getHasActiveCancelledAssignments` as we're working with plural assignments here?